components/prism-factor.js

```diff
@@ -0,0 +1,277 @@
+(function (prism) {
+
+	var factor = {
+
+		comment: { // ! single-line exclamation point comments with whitespace after/around the !
+			pattern: /(^|\s|(?!"\s*))(?:![\t ].*|!$)/,
+			lookbehind: true,
+			inside: {
+				'bold': /\b(?:TODO|FIXME|NOTE|BUG|XXX|HACK)(?:\b|:)/
+			}
+		},
+
+		number: [
+			// basic base 10 integers 9, -9
+			{ 	pattern: /(^|\s)[+-]?\d+(?=\s|$)/,
+				lookbehind: true },
+
+			// base prefix integers 0b010 0o70 0xad 0d10 0XAD -0xa9
+			{ 	pattern: /(^|\s)[+-]?0(?:b[01]+|o[0-7]+|d\d+|x[\dA-F]+)(?=\s|$)/i,
+				lookbehind: true },
+
+			// fractional ratios 1/5 -1/5 and the literal float approximations 1/5. -1/5.
+			{ 	pattern: /(^|\s)[+-]?\d+\/\d+\.?(?=\s|$)/,
+				lookbehind: true },
+
+			// positive mixed numbers 23+1/5 +23+1/5
+			{ 	pattern: /(^|\s)\+?\d+\+\d+\/\d+(?=\s|$)/,
+				lookbehind: true },
+
+			// negative mixed numbers -23-1/5
+			{ 	pattern: /(^|\s)-\d+-\d+\/\d+(?=\s|$)/,
+				lookbehind: true },
+
+			// basic decimal floats -0.01 0. .0 .1 -.1 -1. -12.13 +12.13
+			// and scientific notation with base 10 exponents 3e4 3e-4 .3e-4
+			{ 	pattern: /(^|\s)[+-]?(?:\d*\.\d+|\d+\.\d*|\d+)(?:e[+-]?\d+)?(?=\s|$)/i,
+				lookbehind: true },
+
+			// NAN literal syntax NAN: 80000deadbeef, NAN: a
+			{ 	pattern: /(^|\s)NAN:[\n\t ]+[\da-fA-F]+(?=\s|$)/,
+				lookbehind: true },
+
+			/*
+				base prefix floats 0x1.0p3 (8.0) 0b1.010p2 (5.0) 0x1.p1 0b1.11111111111111111111111111p1111111111111111
+				"The normalized hex form ±0x1.MMMMMMMMMMMMM[pP]±EEEE allows any floating-point number to be specified precisely.
+				The values of MMMMMMMMMMMMM and EEEE map directly to the mantissa and exponent fields of the binary IEEE 754 representation."
+				<https://docs.factorcode.org/content/article-syntax-floats.html>
+			*/
+			{ 	pattern: /(^|\s)[+-]?0(?:b1\.[01]*|o1\.[0-7]*|d1\.\d*|x1\.[\dA-F]*)p\d+(?=\s|$)/i,
+				lookbehind: true }
+		],
+
+		boolean: {
+			pattern: /(^|\s|\r\n)(?:t|f)(?=\s|$)/,
+			lookbehind: true
+		},
+
+		'special-string-literal': {
+			pattern: /(^|\s)[A-Z0-9\-]+"\s(?:\\"|[^"])*"/,
```

Comment:
Yes, whitespace after the opening double-quote is really intentional. 

The words `URL"`, `SBUF"`, and `P"` (path literal) words are the target of `special-string-literal`. They are normal parsing words and must be followed by whitespace, then the content of the string and a closing quote, i.e `URL" https://github.com"`;  ref: [`URL"`](https://docs.factorcode.org/content/word-URL__quo__,urls.html).

<details><summary> Long explanation </summary> 

The double-quote with whitespace (or another string) preceding it, is treated specially by the Factor parser. Usually, as you've seen, tokens must be surrounded by whitespace, but the builtin normal string literal `"like this"` does not have to have whitespace after the first quote-mark (as it would, if `"` was implemented as a normal parsing word that calls `skip-blank` and looks for a closing `"`). Otherwise, builtin strings would be written like `" this"` and the first space after the `"` would be required but dropped by the parser.  </details>

Reply:
It seems to only match the `" \\"` part of `" \\""` though, so that's good.

Reply:
Sorry, `" \\""` is a bad example. Let's take `P" as\\""` which is fully matched.

Reply:
I'm fine with this, because it's hard to do better here than not matching the last erroneous quote mark, without lookaround that I'd rather not use with strings.

On the test page for me, all but the last quote mark in `P" as\\""` is matched.

![Screenshot from 2020-02-09 17-13-07](https://user-images.githubusercontent.com/3238748/74111065-9e6ad800-4b5f-11ea-8fc8-06d090e8668f.png)